File: sistemaFacturacion/src/main/java/prueba/sistemaFacturacion/persistence/entity/Cliente.java
package prueba.sistemaFacturacion.persistence.entity;

import jakarta.persistence.*;
import lombok.*;

@Builder
@Entity
@Table(name = "cliente")
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Cliente {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_nombre", nullable = true)
    private Usuario usuario;

    @Column(name = "identificacion", unique = true, nullable = false, length = 10, updatable = false)
    private String identificacion;

    @Column(name = "telefono")
    private String telefono;

    @Column(name = "direccion")
    private String direccion;
}

File: reactFacturacion/src/interface/cliente.ts
export interface Cliente {
    id: number;
    nombre: string;
    identificacion: string;
    telefono: string;
    correo: string;
    direccion: string;
}

export interface ClienteRequest {
    usuario: string;
    identificacion: string;
    telefono: string;
    direccion: string;
}